include/kernel.h

```diff
@@ -3002,6 +3002,22 @@ static inline bool k_work_pending(struct k_work *work)
 	return atomic_test_bit(work->flags, K_WORK_STATE_PENDING);
 }
 
+/**
+ * @brief      Check if a delayed work item is pending.
+ *
+ * This routine indicates if the work item @a work is pending in a workqueue's
+ * queue or waiting for the delay timeout.
+ *
+ * @param      work  Address of delayed work item.
+ *
+ * @return     true if work item is pending, or fales if it is not pending.
+ */
+static inline bool k_delayed_work_pending(struct k_delayed_work *work)
+{
+	return !z_is_inactive_timeout(work->timeout) ||
+	       k_work_pending(work->work);
```

Comment:
`&work->work`?

Reply:
Thanks. I didn't actually test this. Maybe I should have submitted it as just a draft first.

I was mainly interested to know if the two functions are enough to cover the whole range from submitted to finished executing.